packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/TypeProvider.cs

```diff
@@ -11,6 +11,11 @@ public abstract class TypeProvider
     {
         protected string? _deprecated;
 
+        private string? _filename;
+
+        public string Filename => _filename ??= GetFilename();
+        protected virtual string GetFilename() => $"{Name}.cs";
```

Comment:
Should this be abstract? I'm wondering when we would not need to override it.

Reply:
yeah I prefer it to be abstract - but in Michael's PR it is defined as virtual and I did not change it.

I think abstract is better so that everyone inheriting TypeProvider is forced to assign a name to it.

Reply:
I think we can tackle this in a separate issue then. I will file an issue.